projects/cdk/services/meta-color.service.ts

```diff
@@ -0,0 +1,44 @@
+import type {OnDestroy} from '@angular/core';
+import {inject, Injectable} from '@angular/core';
+import {WA_WINDOW} from '@ng-web-apis/common';
+import {Subject} from 'rxjs';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class TuiMetaColorService implements OnDestroy {
+    private readonly document = inject(WA_WINDOW).document;
+    private readonly value = new Subject<string>();
+
+    protected sub = this.value.subscribe({
+        next: (value) => this.metaThemeColor.setAttribute('content', value),
+    });
+
+    public getColor(): string {
+        return this.metaThemeColor.getAttribute('content') ?? '';
+    }
+
+    public setColor(value: string): [prev: string, next: string] {
+        const prev = this.getColor();
+
+        this.value.next(value);
+
+        return [prev, value];
```

Comment:
I don't think we should return anything. Wherever you call `setColor` you can just call `getColor` yourself and you already have the new value.

Reply:
fixed